arangod/Transaction/Methods.cpp

```diff
@@ -1693,8 +1694,12 @@ Future<OperationResult> transaction::Methods::truncateAsync(std::string const& c
   TRI_ASSERT(_state->status() == transaction::Status::RUNNING);
 
   OperationOptions optionsCopy = options;
-  auto cb = [this, collectionName](OperationResult res) {
+  auto cb = [this, collectionName,
+             handler = rest::RestHandler::CURRENT_HANDLER](OperationResult res) {
+    auto old = rest::RestHandler::CURRENT_HANDLER;
+    rest::RestHandler::CURRENT_HANDLER = handler;
```

Comment:
That was my initial thought too, but it would be a big change to backport, and we would have to do it manually since #12624 has a bunch of other changes we wouldn't want to backport. This approach is pretty minimal, and should work. As long as it's properly fixed in devel going forward, then I don't care too much about a patchwork fix in 3.6 and 3.7.